docs/kb/use-functions.md

```diff
@@ -11,28 +11,76 @@ cta:
     - text: See our complete list of prebuilt functions - [here](/examples)
     - text: Learn how to write your own function - [here](contribute-functions).
 ---
-
 # Getting started with using functions
 
 The steps below describe a generalized deployment step of a function on the VMware Event Broker Appliance. For customers looking to get started quickly, please look at deploying from our growing list of [Prebuilt Functions](/examples). The functions are organized by the language that they are written in and have well-documented README.md files with detailed deployment steps.
 
 Deployment or development of functions can require the use of the following development tools: `git`, `docker`, and `kubectl`.  If you would like some help in how to setup your workstation to use those tools, an in-depth tutorial is available that shows how to do that and to modify and deploy the [kn-ps-slack](https://github.com/vmware-samples/vcenter-event-broker-appliance/tree/master/examples/knative/powershell/kn-ps-slack) function (Knative powershell slack notification function) as an example.  The tutorial can be found here: [in-depth function tutorial](function-tutorial-intro).
 
+### Deployment Prerequisites
+
+Before proceeding to deploy a function, you must have VMware Event Broker Appliance deployed and be able to SSH to the appliance or have access to the kubernetes configuration file (`/root/.kube/config`) locally on your desktop to authenticate.
+
+### Function Deployment with Kubernetes Secrets
+
+Most functions require configuration parameters to work correctly. A function author can hard code parameter values inside the function. However, any changes to the configuration parameters means the function's container image must be rebuilt. By using configuration parameters stored in a Kubernetes secret, a function can be reconfigured by simply changing the secret. 
+
+For most of this walk-through, the [`kn-ps-echo`](/examples) function from the examples folder is used. The `kn-ps-echo` function does not use secrets because the function simply echoes any event that it receives. 
```

Comment:
nit: I'd completely remove echo from this section. After you restructure, we already cover echo in the section earlier when describing how to deploy a simple function.